alchemy/test/cloudflare/redirect-rule.test.ts

```diff
@@ -0,0 +1,232 @@
+import { describe, expect } from "vitest";
+import { alchemy } from "../../src/alchemy.ts";
+import { createCloudflareApi } from "../../src/cloudflare/api.ts";
+import { RedirectRule } from "../../src/cloudflare/redirect-rule.ts";
+import { Zone } from "../../src/cloudflare/zone.ts";
+import { destroy } from "../../src/destroy.ts";
+import { BRANCH_PREFIX } from "../util.ts";
+
+import "../../src/test/vitest.ts";
+
+const api = await createCloudflareApi();
+
+const test = alchemy.test(import.meta, {
+  prefix: BRANCH_PREFIX,
+});
+
+describe("RedirectRule", () => {
+  // Use BRANCH_PREFIX for deterministic, non-colliding test resources
+  const testDomain = `${BRANCH_PREFIX}-redirect-test.dev`;
+
+  test("create, update, and delete wildcard redirect rule", async (scope) => {
+    let zone: Zone | undefined;
+    let redirectRule: RedirectRule | undefined;
+
+    try {
+      // Create a test zone first
+      zone = await Zone(`${testDomain}-zone`, {
+        name: testDomain,
+        type: "full",
+        jumpStart: false,
+      });
+
+      // Create a wildcard redirect rule
+      redirectRule = await RedirectRule(`${BRANCH_PREFIX}-wildcard-redirect`, {
+        zone: zone.id,
+        requestUrl: `https://*.${testDomain}/files/*`,
+        targetUrl: `https://${testDomain}/\${1}/files/\${2}`,
+        statusCode: 301,
+        preserveQueryString: true,
+      });
+
+      expect(redirectRule.ruleId).toBeTruthy();
+      expect(redirectRule.rulesetId).toBeTruthy();
+      expect(redirectRule.zoneId).toEqual(zone.id);
+      expect(redirectRule.requestUrl).toEqual(
+        `https://*.${testDomain}/files/*`,
+      );
+      expect(redirectRule.targetUrl).toEqual(
+        `https://${testDomain}/\${1}/files/\${2}`,
+      );
+      expect(redirectRule.statusCode).toEqual(301);
+      expect(redirectRule.preserveQueryString).toEqual(true);
+      expect(redirectRule.enabled).toEqual(true);
+
+      // Verify the rule was created by querying the API directly
+      const getRuleResponse = await api.get(
+        `/zones/${zone.id}/rulesets/${redirectRule.rulesetId}/rules/${redirectRule.ruleId}`,
+      );
+      expect(getRuleResponse.status).toEqual(200);
+
+      const ruleData: any = await getRuleResponse.json();
+      expect(ruleData.result.action).toEqual("redirect");
+      expect(ruleData.result.enabled).toEqual(true);
```

Comment:
Ideally we would actually invoke the URL and test it redirects intead of just asserting it exists. Check behavior.